### ISSUE TITLE
Support for half-float rendertarget

### ISSUE THREAD

**Issue by tsone** [mrdoob/three.js#5973]
Half float may be faster on some hardware and it occupies less space than regular float rendertargets. It also looks like the upcoming WebGL 2.0 supports it. I would add a new `THREE.HalfFloatType` and modify `WebGLRenderTarget` to accept it.

New extensions to be enabled:
```
EXT_color_buffer_half_float
OES_texture_half_float
OES_texture_half_float_linear
```

How about it?

**Comment by mrdoob** [comment]
Sounds good to me. The enabling extensions approach we have now worry me a bit thought. Does it make the init take longer?

**Comment by tsone** [comment]
I'd guess the delay would be the same as for enabling float textures -- or less. I didn't try it yet.

So there would be a warning if half-floats are not supported...

**Comment by tsone** [comment]
Added pull request #5975

**Comment by tsone** [comment]
Closed because #5975 is merged.
